doc/NotaryContentTrust.md

```diff
@@ -91,7 +91,7 @@ Notary client directory exists in `~/.notary`. A configuration file has be creat
 ```
 `trust_dir` is the location where the trust directory gets initialized and downloaded. `remote_server` has a sub field `url` which configures the URL of the registry server host name. 
 
-Notary needs authorization credentials to communicate to the Notary Server. For this, a Service Principal for the Azure Container Registry with Owner and Push access must be created. Follow instructions in [this](https://docs.microsoft.com/en-us/azure/container-registry/container-registry-auth-service-principal) link to create a Service Principal. It is mandatory to create the Service Principals to use the Content Trust feature of Azure Container Registry. Read [this](https://docs.microsoft.com/en-us/azure/iot-edge/production-checklist?view=iotedge-2020-11#security-considerations) documentation of IoT Edge on how to use Service Principals.
+Notary needs authorization credentials to communicate to the Notary Server. For this, a Service Principal for the Azure Container Registry with Owner, AcrImageSigner and AcrPush access must be created. Follow instructions in [this](https://docs.microsoft.com/en-us/azure/container-registry/container-registry-auth-service-principal) link to create a Service Principal. It is mandatory to create the Service Principals to use the Content Trust feature of Azure Container Registry. Read [this](https://docs.microsoft.com/en-us/azure/iot-edge/production-checklist?view=iotedge-2020-11#security-considerations) documentation of IoT Edge on how to use Service Principals.
```

Comment:
If the SP has AcrImageSigner and AcrPush rights, does it still need Owner?

Reply:
Yes In one of the steps to to rotate keys, it was failing. Once I added this role - it was successful.

Reply:
Owner role for setting the trust collection, signer role for key rotation and Push role for pushing image to repo

Reply:
Owner gives the SP rights to do _anything_, including assign other roles, so it's not a secure option. I'm wondering if the notary client can set up the TUF collection with a lesser access role? Have you tried granting _just_ AcrImageSigner and AcrPush (not Owner)? [Some documentation](https://docs.microsoft.com/en-us/azure/container-registry/container-registry-content-trust#grant-image-signing-permissions) I found only mentioned AcrImageSigner and AcrPush.

Reply:
Let me try without Owner.

Reply:
I tried without the owner role and while notary key rotation - it says the following error

Reply:
> For making any changes in the signature collection, Owner role is essential

Did you write that, or was it part of the error message?

If that's not part of the error message, then please try with at least Contributor instead of Owner. If that doesn't work, we can keep Owner in the doc, but I'm hoping we don't have to.

Reply:
If we only need Owner/Contributor for key rotation, perhaps we should tell the user to create 3 service principals for each role in the process, Owner for Step 2, SignImage/Push for Step 3. Pull for Step 4?

Reply:
> > For making any changes in the signature collection, Owner role is essential
> 
> Did you write that, or was it part of the error message?
> 
> If that's not part of the error message, then please try with at least Contributor instead of Owner. If that doesn't work, we can keep Owner in the doc, but I'm hoping we don't have to.

I did write that and it was not part of the message. I tried with multiple service principals. Here is the result
1. Only owner - Fails to initialize the trust collection i.e. Notary init fails
2. Owner, AcrPush - Key rotate fails but can initialize the trust collection. 
3. AcrPush, AcrImageSigner - Key rotate fails but can initialize the trust collection. 
4. Owner, AcrPush, AcrImageSigner - Can initialize and also rotate the key. 
5. Owner, AcrImageSigner - cannot initialize the trust collection

So in conclusion, we need all three as minimum to do the operation. @darobs I am not sure which role plays what so I will leave only the generic instructions on which roles are needed to get started.

Reply:
@ggjjj Were you able to assign & use Contributor?

Reply:
RE: the 3 service principals.  I was hoping to model each one as the roles needed to use this feature.  We would expect the first user (a person as an admin/devops) to have the ability to install certificates and rotate keys - they would have Owner, AcrPush, AcrImageSigner permissions.  The second user is an image builder (probably automated) that would be the one that pushes and signs the image - they would have AcrPush, AcrImageSigner permissions. Third, there would be a user of that image (like IoT Edge) that pulls it.  This may be slightly out of scope for this document, but if it doesn't muck up the flow too much, it would be a nice addition.

Reply:
Contributor, AcrPush, AcrImageSigner works